src/GTP.cpp

```diff
@@ -172,6 +176,13 @@ const std::string GTP::s_commands[] = {
     "heatmap",
     "lz-analyze",
     "lz-genmove_analyze",
+    "lz-estimatememory",
+    "lz-setoption",
+    ""
+};
+
+const std::string GTP::s_options[] = {
+    "max_memory <total memory in MB> <optional: fraction for Neural Network Cache>- set the maximum memory (in MB) leelaz should utilize. Optionally set fraction of memory remaining for the tree search which should be used by Neural Network cache (by default 0.1)",
```

Comment:
For the working of lz-setoption, can you take a look at the UCI spec (for chess engines) and see how setoption works there? I'd prefer it if lz-setoption were to follow the same format, but perhaps adding float support.

It doesn't provide any way to provide a description to the GUI, though (but the options can be named more descriptively!), but I think that's still better than this format which is OK for someone working in the terminal, or someone who adapts their GUI specifically to LZ's idiosyncrasies, but not really where we want to go for Go engines in general.

So a session would look like:
```
lz-setoption
option name Maximum Memory Use (MiB) type spin default 2048 min 128 max 131072
option name Percentage of memory for cache type spin default 10 min 1 max 99

lz-setoption name Maximum Memory Use (MiB) value 1024
lz-setoption name Percentage of memory for cache value 20
```